app/queries/keyword_search_entries_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class KeywordSearchEntriesQuery
+  def initialize(relation = KeywordSearchEntry)
+    @relation = relation
+  end
+
+  def with_keyword(keyword_id)
+    @relation = relation.where(keyword_id: keyword_id)
+
+    self
+  end
+
+  def top_ads_count
+    relation.where(kind: :ads, position: :top).size
+  end
+
+  def total_ads_count
+    relation.where(kind: :ads).size
+  end
+
+  def non_ads_count
+    relation.where(kind: :non_ads).size
+  end
+
+  def top_ads_urls
+    urls_of(relation.where(kind: :ads, position: :top))
```

Comment:
is it duplicated with the KeywordDecorator.rb file there 🤔

Reply:
My bad. I forgot to remove that file. Fixed in 6310a736cc744121a554172071b42970e95a8e83